### ISSUE TITLE
Error message overlap

### ISSUE THREAD

**Issue by Alex-PK** [rust-lang/rust#37075]
When I have few errors (probably only where there is just one) the "running" error messages and the (same) messages after "waiting for other jobs to finish" are overlapping. Here is an example:

```
errorerror: expected `,` or `>` after lifetime name, found `T`: expected `,` or `>` after lifetime name, found `T`

      --> --> src/lib.rs:266:36src/lib.rs:266:36

    ||

266266  | | impl<'a, 'b, T> ::std::ops::Add<'b impl<'a, 'b, T> ::std::ops::Add<'b T> for &'a ValueT
> for &'a Value
        | |                                                                       ^^

        ||

notenote: did you mean a single argument type &'a Type, or did you mean the comma-separated arguments 'a, Type?: did you mean a single argument type &'a Type, or did you mean the comma-separated arguments 'a, Type?

      --> --> src/lib.rs:266:33src/lib.rs:266:33

        ||

266266  | | impl<'a, 'b, T> ::std::ops::Add<impl<'a, 'b, T> ::std::ops::Add<'b T>'b T> for &'a Value for &'a Value

    | |                                                                 ^^^^^^^^^^



errorerror: aborting due to previous error: aborting due to previous error



Build failed, waiting for other jobs to finish...
```

If I run the same command again, the messages are overlapping in different ways.

It looks like 2 threads are printing at the same time.

This happens on rust 1.12 on Linux with the new error messages format, but also on older versions with the old format. If I remember correctly, I saw it on OSX too, but I can't check right now.

Sorry if this was already reported, but I couldn't find it in the issues list, as I don't know exactly what terms to use for the search.

**Comment by Stebalien** [comment]
See https://github.com/rust-lang/rust/issues/32486

**Comment by Mark-Simulacrum** [comment]
So I personally have never seen this. It's possible this is because I rarely see errors from two separate rustc's running in parallel; @alexcrichton you closed #32486 as "implemented" -- do you agree that we should close this as well?

**Comment by alexcrichton** [comment]
Indeed!
